// @percy/config/dist/migrate.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.addMigration = addMigration;
exports.clearMigrations = clearMigrations;
exports.default = migrate;

var _normalize = _interopRequireDefault(require("./normalize"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// Global set of registered migrations
const migrations = new Set(); // Register a migration function

function addMigration(migration) {
  migrations.add(migration);
} // Clear all migration functions


function clearMigrations() {
  migrations.clear();
} // Assigns a value to the object at the path creating any necessary nested
// objects along the way


function assign(obj, path, value) {
  path.split('.').reduce((loc, key, i, a) => loc[key] = i === a.length - 1 ? value : loc[key] || {}, obj);
  return obj;
} // Calls each registered migration function with a normalize provided config
// and a `set` function which assigns values to the returned output


function migrate(config) {
  let output = {
    version: 2
  };
  let input = (0, _normalize.default)(config);
  let set = assign.bind(null, output);
  migrations.forEach(migration => {
    migration(input, set);
  });
  return output;
}
