// @percy/cli-upload/dist/config.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.migration = migration;
exports.schema = void 0;
const schema = {
  upload: {
    type: 'object',
    additionalProperties: false,
    properties: {
      files: {
        anyOf: [{
          type: 'string'
        }, {
          type: 'array',
          items: {
            type: 'string'
          }
        }],
        default: '**/*.{png,jpg,jpeg}'
      },
      ignore: {
        anyOf: [{
          type: 'string'
        }, {
          type: 'array',
          items: {
            type: 'string'
          }
        }],
        default: ''
      }
    }
  }
};
exports.schema = schema;

function migration(input, set) {
  /* eslint-disable curly */
  if (input.version < 2) {
    var _input$imageSnapshots, _input$imageSnapshots2;

    // image-snapshots and options were renamed
    if (((_input$imageSnapshots = input.imageSnapshots) === null || _input$imageSnapshots === void 0 ? void 0 : _input$imageSnapshots.files) != null) set('upload.files', input.imageSnapshots.files);
    if (((_input$imageSnapshots2 = input.imageSnapshots) === null || _input$imageSnapshots2 === void 0 ? void 0 : _input$imageSnapshots2.ignore) != null) set('upload.ignore', input.imageSnapshots.ignore);
  }
}
