// @percy/cli-snapshot/dist/config.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.migration = migration;
exports.schema = void 0;
const schema = {
  static: {
    type: 'object',
    additionalProperties: false,
    properties: {
      baseUrl: {
        type: 'string',
        default: '/'
      },
      files: {
        anyOf: [{
          type: 'string'
        }, {
          type: 'array',
          items: {
            type: 'string'
          }
        }],
        default: '**/*.{html,htm}'
      },
      ignore: {
        anyOf: [{
          type: 'string'
        }, {
          type: 'array',
          items: {
            type: 'string'
          }
        }],
        default: ''
      }
    }
  }
};
exports.schema = schema;

function migration(input, set) {
  /* eslint-disable curly */
  if (input.version < 2) {
    var _input$staticSnapshot, _input$staticSnapshot2, _input$staticSnapshot3;

    // static-snapshots and options were renamed
    if (((_input$staticSnapshot = input.staticSnapshots) === null || _input$staticSnapshot === void 0 ? void 0 : _input$staticSnapshot.baseUrl) != null) set('static.baseUrl', input.staticSnapshots.baseUrl);
    if (((_input$staticSnapshot2 = input.staticSnapshots) === null || _input$staticSnapshot2 === void 0 ? void 0 : _input$staticSnapshot2.snapshotFiles) != null) set('static.files', input.staticSnapshots.snapshotFiles);
    if (((_input$staticSnapshot3 = input.staticSnapshots) === null || _input$staticSnapshot3 === void 0 ? void 0 : _input$staticSnapshot3.ignoreFiles) != null) set('static.ignore', input.staticSnapshots.ignoreFiles);
  }
}
